app/src/main/java/com/swent/assos/ui/screens/QrCode.kt

```diff
@@ -1,9 +1,143 @@
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.swent.assos.R
+import com.swent.assos.model.data.Event
+import com.swent.assos.model.view.EventViewModel
+import com.swent.assos.ui.theme.Purple40
+import com.swent.assos.ui.theme.Purple80
 
 @Composable
 fun QrCode() {
-  Text("QrCode")
+  val eventViewModel: EventViewModel = hiltViewModel()
+  val events by eventViewModel.allEvents.collectAsState()
+  Scaffold(
+    topBar = {
+      Column(modifier = Modifier.fillMaxWidth()) {
+        Row(
+          modifier =
+          Modifier
+            .padding(8.dp)
+            .align(Alignment.CenterHorizontally)
+            .wrapContentHeight(align = Alignment.CenterVertically)) {
+          Text(
+            text = "Student",
+            style =
+            TextStyle(
+              fontSize = 35.sp,
+              fontFamily = FontFamily(Font(R.font.impact)),
+              fontWeight = FontWeight(400),
+              color = Purple80))
+          Text(
+            text = "Sphere",
+            style =
+            TextStyle(
+              fontSize = 35.sp,
+              fontFamily = FontFamily(Font(R.font.impact)),
+              fontWeight = FontWeight(400),
+              color = Purple40))
+        }
+      }
+    }) { paddingValues ->
+    Surface(color = MaterialTheme.colorScheme.background) {
+      LazyColumn(
+        modifier = Modifier
+          .fillMaxSize()
+          .padding(paddingValues)
+          .padding(16.dp),
+        userScrollEnabled = true
+      ) {
+        items(events) {
+          TicketItem(event = it)
+        }
+      }
+    }
+  }
 }
+
+@Composable
+fun TicketItem(event: Event) {
+  var isExpanded by remember { mutableStateOf(false) } // Keeps track of the expanded state
+
+  Card(
+    modifier = Modifier
+      .fillMaxWidth()
+      .padding(vertical = 8.dp)
+      .clickable { isExpanded = !isExpanded }, // Toggle the expanded state on click
+    shape = RoundedCornerShape(8.dp),
+    //elevation = 2.dp
+  ) {
+    Column {
+      Row(
+        modifier = Modifier
+          .fillMaxWidth()
+          .padding(16.dp),
+        verticalAlignment = Alignment.CenterVertically
+      ) {
+        Column(
+          modifier = Modifier
+            .weight(1f)
+            .padding(end = 16.dp)
+        ) {
+          Text(
+            text = "Ticket ${event.title}",
+            style = MaterialTheme.typography.titleMedium
+          )
+          Spacer(modifier = Modifier.height(4.dp))
+          Text(
+            text = event.date,
+            style = MaterialTheme.typography.bodyMedium
+          )
+        }
+        // You can place an icon indicating the expanded state here
+      }
```

Comment:
Here, you have a column in your row and it's the single element in it. You do not need to create the row.